Automattic/wp-calypso
# Odyssey Stats: migrate Odyssey Stats to TypeScript

Related to #76570 

## Proposed Changes

* Adds `tsconfig.json`
* Migrated all code to TypeScript except for `apps/odyssey-stats/src/lib/create-odyssey-config.js` which would need some major refactoring

Two TS checks still failing, which we are not going to fix are:

- `context.hash` is used as object everywhere, but which has the type of a string: https://github.com/Automattic/wp-calypso/pull/76619/files#diff-bce221d5a6ad0e844fd9d029b0277f265484b453a1b615581140ef6efbe0c2f7R10
- `currentQuery` of `RouteProvider` is defaulted to `false` which is determined as boolean in TS and wouldn't allow object. https://github.com/Automattic/wp-calypso/pull/76619/files#diff-f275f3d4d16209e6b6eadd9ebc957149927ac126831ddf351fcf74596e34cbcfR34


## Testing Instructions

* Ensure all checks successful on GitHub
* Build it locally
* Ensure it still builds and works well

## Pre-merge Checklist
